rfcs/0000-top-level-assumptions.md

```diff
@@ -172,12 +172,12 @@ The different `loose` or `loose`-like options we currently have in our plugins m
-| `noCallClass` | Assume that classes are always instantiated with `new` and that the code never tries to call them as function. This lets Babel skipping the `this instanceof ThisClass` check (`_classCallCheck`). | `loose` in `transform-classes` | |
+| `noClassCalls` | Assume that classes are always instantiated with `new` and that the code never tries to call them as function. This lets Babel skipping the `this instanceof ThisClass` check (`_classCallCheck`). | `loose` in `transform-classes` | |
 | `noDocumentAll` | When compiling the `??` and `?.` operators, assume that they are never used with `document.all` and thus `== null` is safe | `loose` in `proposal-optional-chaining` and `proposal-nullish-coalescing-operator` | [🔗](https://babeljs.io/repl#?browsers=chrome%2070&build=&builtIns=false&spec=false&loose=false&code_lz=IYfgdARg3AUMAEIT2kA&debug=false&forceAllTransforms=false&shippedProposals=true&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=script&lineWrap=true&presets=env%2Cenv&prettier=false&targets=&version=7.10.1&externalPlugins=) | 3 |
 | `pureGetters` | When an expression that might invoke getters needs to be evaluated multiple times, Babel caches its value. When this option is enabled, the expression can be safely re-evaluated. For example, `a.b?.()` can be compiled to `a.b != null && a.b()` without caching `a.b` | `loose` in `proposal-optional-chaining` and in `proposal-object-rest-spread` _(not documented)_ | [🔗](https://babeljs.io/repl#?browsers=chrome%2070&build=&builtIns=false&spec=false&loose=true&code_lz=IYOgRg_CAUCUQ&debug=false&forceAllTransforms=false&shippedProposals=true&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=script&lineWrap=true&presets=env&prettier=false&targets=&version=7.10.1&externalPlugins=) |
 | `enumerableModuleMeta` | When compiling ESM to CJS, Babel defines a non-enumerable, non-witable, non-configurable `__esModule` property on the `exports` object. When this option is enabled, that property is set using a simple assignment | `loose` in `transform-modules-commonjs` |
 | `constantReexports` | When re-exporting an imported value, assume that it's value doesn't change and set it with a simple assignment | `loose` in `transform-modules-commonjs`, `transform-modules-umd`, `transform-modules-amd` _(not documented)_ | | |
-| `newableArrowFunctions` | Do not check that arrow functions are never instantiated using `new` | `spec` in `transform-arrow-functions` | | 4 |
+| `noNewArrows` | Assume that the code never tries to instantiatearrow function using `new`, so Babel can avoid injecting checks to prevent it | `spec` in `transform-arrow-functions` | | 4 |
```

Comment:
```suggestion
| `noNewArrows` | Assume that the code never tries to instantiate arrow function using `new`, so Babel can avoid injecting checks to prevent it | `spec` in `transform-arrow-functions` | | 4 |
```

what about concise methods, generators, other things that lack `.prototype`?

Reply:
This is definitely something that we should add (for example, I mentioned `noNewMethods` in https://github.com/babel/rfcs/pull/5#discussion_r567348593).
However, with this RFC I'm only trying to specify _existing_ assumptions of the `loose`/`spec` options, and checking for generators/methods/etc is not something that Babel does yet.